### PR TITLE
Remove HttpRequest serialization test

### PR DESCRIPTION
**Pull request by danmosemsft**
Fixes https://github.com/dotnet/corefx/issues/14068
I suspect https://github.com/dotnet/corefx/pull/14175 fixed this and I forgot to enable it. It passes now.

@Priya91

### CODE REVIEW

**Comment by stephentoub** [comment]
@dotnet-bot test Outerloop Windows_NT Debug please
@dotnet-bot test Outerloop Ubuntu14.04 Debug please
@dotnet-bot test Outerloop OSX Debug please

**Comment by danmosemsft** [comment]
Forgot outerloop - thanks

**Comment by danmosemsft** [comment]
Aw nuts, that means I hadn't run it and it doesn't pass. Abort.

**Comment by danmosemsft** [comment]
Since I eliminated serialization in my original fix (see notes in PR) this test should actually be deleted.

@dotnet-bot test Outerloop Windows_NT Debug please
@dotnet-bot test Outerloop Ubuntu14.04 Debug please
@dotnet-bot test Outerloop OSX Debug please

**Comment by danmosemsft** [comment]
https://github.com/dotnet/corefx/pull/14175/files added a test to make sure it fails. That one remains

File: src/System.Net.Requests/tests/System.Net.Requests.Tests.csproj [hunk 1]
```diff
@@ -47,7 +47,6 @@
     <Compile Include="HttpWebRequestHeaderTest.cs" />
     <Compile Include="HttpWebRequestTest.netstandard.cs" />
     <Compile Include="HttpWebResponseHeaderTest.cs" />
-    <Compile Include="HttpRequestResponseSerializationTest.cs" />
     <Compile Include="$(CommonTestPath)\System\Runtime\Serialization\Formatters\BinaryFormatterHelpers.cs">
```
Comment (Priya91) [inline src/System.Net.Requests/tests/System.Net.Requests.Tests.csproj]: We should remove the BinaryFormatterHelpers.cs from csproj as well, suspect it's used elsewhere other than the removed file. Don't require it in compilation.

**Comment by danmosemsft** [comment]
@dotnet-bot help please

**Pull request merged**